connect/src/indexers/AlchemyClient.ts

```diff
@@ -43,35 +46,83 @@ class AlchemyClient {
     return endpoint !== undefined;
   }
 
+  private async batchFetchFromAlchemy(
+    endpoint: string,
+    requests: Array<{ method: string; params: any[]; id: number }>,
+    signal?: AbortSignal,
+  ): Promise<Array<{ id: number; result: any }>> {
+    const baseUrl = `https://${endpoint}.g.alchemy.com/v2/${this.key}`;
+
+    const response = await fetch(baseUrl, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(
+        requests.map((req) => ({
+          jsonrpc: "2.0",
+          id: req.id,
+          method: req.method,
+          params: req.params,
+        })),
+      ),
+      signal,
+    });
+
+    const data = await response.json();
+    return data;
+  }
+
+  private parseEthTokenResponse(tokenBalances: any[], bals: Balances): void {
+    for (let item of tokenBalances) {
+      const bal = parseBalance(item.tokenBalance || item.balance);
+      if (bal !== null) {
+        bals[item.contractAddress] = bal;
+      }
+    }
+  }
+
+  private parseEthNativeResponse(nativeBalance: string, bals: Balances): void {
+    const nativeBal = parseBalance(nativeBalance);
+    if (nativeBal !== null) {
+      bals.native = nativeBal;
+    }
+  }
+
   async getBalances(
     network: Network,
     chain: Chain,
     walletAddr: string,
     signal?: AbortSignal,
   ): Promise<Balances> {
     const endpoint = ALCHEMY_CHAINS[network][chain];
-    if (!endpoint) throw new Error("Chain not supported by Alchemy indexer");
```

Comment:
This chunk of code just extracted to a function with body's params/methods passed in